pkg/devfile/parser/parse.go

```diff
@@ -90,83 +102,66 @@ func ParseFromData(data []byte) (d DevfileObj, err error) {
 		return d, err
 	}
 
-	return parseDevfile(d)
+	return parseDevfile(d, true)
 }
 
-func parseParent(d DevfileObj) error {
-	parent := d.Data.GetParent()
-
-	parentData, err := ParseFromURL(parent.Uri)
-	if err != nil {
-		return err
-	}
-	klog.V(4).Infof("overriding data of devfile with URI: %v", parent.Uri)
+func parseParentAndPlugin(d DevfileObj) (err error) {
+	flattenedParent := &v1.DevWorkspaceTemplateSpecContent{}
+	if d.Data.GetParent() != nil {
+		if !reflect.DeepEqual(d.Data.GetParent(), &v1.Parent{}) && d.Data.GetParent().Uri != "" {
+			parent := d.Data.GetParent()
 
-	// override the parent's components, commands, projects and events
-	err = parentData.OverrideComponents(d.Data.GetParent().Components)
-	if err != nil {
-		return err
-	}
+			parentData, err := ParseFromURL(parent.Uri)
+			if err != nil {
+				return err
+			}
```

Comment:
We should not check for only URI. We should accept all. But since we dont have the implementation for it, we can have a switch case and err out for the other cases, saying its not supported yet. I think thats neater.

Reply:
Lets also rename the var from `parentData` to `parentDevfileObj` or something. It's very confusing in odo and our library repo to read `parentData.Data.GetDevfileWorkspace()` for example